hive-metastore/src/main/java/org/apache/iceberg/hive/legacy/LegacyHiveTableUtils.java

```diff
@@ -56,7 +56,26 @@ static Schema getSchema(org.apache.hadoop.hive.metastore.api.Table table) {
     String schemaStr = props.get("avro.schema.literal");
     Schema schema;
     if (schemaStr != null) {
-      schema = AvroSchemaUtil.toIceberg(new org.apache.avro.Schema.Parser().parse(schemaStr));
+      org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(schemaStr);
+      org.apache.avro.Schema hiveSchema = LegacyHiveSchemaUtils.convertHiveSchemaToAvro(table);
+      org.apache.avro.Schema avroSchemaWithoutNullable =
+          LegacyHiveSchemaUtils.extractActualTypeIfFieldIsNullableTypeRecord(avroSchema);
+      org.apache.avro.Schema hiveSchemaWithoutNullable =
+          LegacyHiveSchemaUtils.extractActualTypeIfFieldIsNullableTypeRecord(hiveSchema);
+
+      org.apache.avro.Schema tableSchema = avroSchema;
+      boolean isHiveSchemaEvolved =
+          LegacyHiveSchemaUtils.isRecordSchemaEvolved(avroSchemaWithoutNullable, hiveSchemaWithoutNullable);
```

Comment:
Why are we describing this in terms of `evolution`? We are simply comparing a Hive schema to a Supplemental schema. They either match or not match (according to some criteria that is defined in the method). Otherwise, there is no `evolution`, `new`, `old`, etc.